extensions-contrib/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureStorageDruidModuleTest.java

```diff
@@ -101,6 +126,108 @@ public void test_getAzureStorageContainer_expectedClient()
+    Assert.assertNotNull(object1);
+    Assert.assertNotNull(object2);
+    Assert.assertNotSame(object1, object2);
+  }
+
+  @Test
+  public void test_getAzureEntityFactory_canCreateAzureEntity()
+  {
+    final Properties props = new Properties();
+    props.put("druid.azure.account", AZURE_ACCOUNT_NAME);
+    props.put("druid.azure.key", AZURE_ACCOUNT_KEY);
+    props.put("druid.azure.container", AZURE_CONTAINER);
+
+    EasyMock.expect(cloudObjectLocation1.getBucket()).andReturn(AZURE_CONTAINER);
+    EasyMock.expect(cloudObjectLocation2.getBucket()).andReturn(AZURE_CONTAINER);
+    EasyMock.expect(cloudObjectLocation1.getPath()).andReturn(PATH);
+    EasyMock.expect(cloudObjectLocation2.getPath()).andReturn(PATH);
+    replayAll();
+
+    injector = makeInjectorWithProperties(props);
+    AzureEntityFactory factory = injector.getInstance(AzureEntityFactory.class);
+    Object object1 = factory.create(cloudObjectLocation1);
+    Object object2 = factory.create(cloudObjectLocation2);
+    Assert.assertNotNull(object1);
+    Assert.assertNotNull(object2);
+    Assert.assertNotSame(object1, object2);
+  }
+
+  @Test
+  public void test_getAzureCloudBlobIteratorFactory_canCreateAzureCloudBlobIterator()
+  {
+    final Properties props = new Properties();
+    props.put("druid.azure.account", AZURE_ACCOUNT_NAME);
+    props.put("druid.azure.key", AZURE_ACCOUNT_KEY);
+    props.put("druid.azure.container", AZURE_CONTAINER);
+    injector = makeInjectorWithProperties(props);
+    AzureCloudBlobIteratorFactory factory = injector.getInstance(AzureCloudBlobIteratorFactory.class);
+    Object object1 = factory.create(EMPTY_PREFIXES_ITERABLE, 10);
+    Object object2 = factory.create(EMPTY_PREFIXES_ITERABLE, 10);
+    Assert.assertNotNull(object1);
+    Assert.assertNotNull(object2);
+    Assert.assertNotSame(object1, object2);
+  }
+
+  @Test
+  public void test_getAzureCloudBlobIterableFactory_canCreateAzureCloudBlobIterable()
+  {
+    final Properties props = new Properties();
+    props.put("druid.azure.account", AZURE_ACCOUNT_NAME);
+    props.put("druid.azure.key", AZURE_ACCOUNT_KEY);
+    props.put("druid.azure.container", AZURE_CONTAINER);
+    injector = makeInjectorWithProperties(props);
+    AzureCloudBlobIterableFactory factory = injector.getInstance(AzureCloudBlobIterableFactory.class);
+    AzureCloudBlobIterable object1 = factory.create(EMPTY_PREFIXES_ITERABLE, 10);
+    AzureCloudBlobIterable object2 = factory.create(EMPTY_PREFIXES_ITERABLE, 10);
+    Assert.assertNotNull(object1);
+    Assert.assertNotNull(object2);
+    Assert.assertNotSame(object1, object2);
+  }
+
+  @Test
+  public void test_getListBlobItemDruidFactory_canCreateListBlobItemDruid()
+  {
+    final Properties props = new Properties();
+    props.put("druid.azure.account", AZURE_ACCOUNT_NAME);
+    props.put("druid.azure.key", AZURE_ACCOUNT_KEY);
+    props.put("druid.azure.container", AZURE_CONTAINER);
+    injector = makeInjectorWithProperties(props);
```

Comment:
I see this duplicated in a lot of tests, maybe we should put it in it's own utility function?